src/Data/ArrayBuffer/Typed.js

```diff
@@ -1,75 +1,219 @@
 "use strict";
 
-// module Data.ArrayBuffer.Typed
 
-exports.asInt8Array = function(v) {
-  return new Int8Array(v.buffer, v.byteOffset, v.byteLength);
-}
+exports.polyFill = function polyFill () {
+    var typedArrayTypes =
+        [ Int8Array, Uint8Array, Uint8ClampedArray, Int16Array
+        , Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array
+        ];
 
-exports.asInt16Array = function(v) {
-  return new Int16Array(v.buffer, v.byteOffset, v.byteLength >>> 1);
-}
+    for (var k in typedArrayTypes) {
+        for (var v in Array.prototype) {
+            if (Array.prototype.hasOwnProperty(v) && !typedArrayTypes[k].prototype.hasOwnProperty(v))
+                typedArrayTypes[k].prototype[v] = Array.prototype[v];
+        }
+    }
+};
 
-exports.asInt32Array = function(v) {
-  return new Int32Array(v.buffer, v.byteOffset, v.byteLength >>> 2);
-}
 
-exports.asUint8Array = function(v) {
-  return new Uint8Array(v.buffer, v.byteOffset, v.byteLength);
-}
+// module Data.ArrayBuffer.Typed
+
+exports.buffer = function buffer (v) {
+    return v.buffer;
+};
+
+exports.byteOffset = function byteOffset (v) {
+    return v.byteOffset;
+};
+
+exports.byteLength = function byteLength (v) {
+    return v.byteLength;
+};
+
+exports.lengthImpl = function lemgthImpl (v) {
+    return v.length;
+};
+
+
+// Typed Arrays
+
+exports.newUint8ClampedArray = function newUint8ClampedArray (a,mb,mc) {
```

Comment:
What about something like:

```
function newArray(f, a, mb, mc) {
    return mc === null ? ( mb === null ? new f(a)
                                       : new f(a,mb)
                         )
                       : new f(a,mb,mc);
}

exports.newUint8ClampedArray = function newUint8ClampedArray (a,mb,mc) {
  return newArray(Uint8ClampedArray, a, mb, mc);
};

```

Reply:
Oh jeez, sorry. I thought JavaScript would be context-sensitive with `new`, and attempt to find a JS object constructor `f` or something... Shoot. Here let me fix this, one sec.